src/components/3d/SimpleModelView.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import {
+  Animated,
+  PanResponder,
+  StyleSheet,
+  View,
+  ViewStyle,
+} from 'react-native';
+import { WebView } from 'react-native-webview';
+import styled from 'styled-components';
+import { ImgixImage } from '@rainbow-me/images';
+import { padding, position } from '@rainbow-me/styles';
+
+export type ModelViewerProps = {
+  readonly setLoading: (loading: boolean) => void;
+  readonly loading: boolean;
+  readonly style?: ViewStyle;
+  readonly uri?: string;
+  readonly alt?: string;
+  readonly fallbackUri?: string;
+};
+
+const StyledWebView = styled(WebView)`
+  ${position.size('100%')};
+`;
+
+const ProgressIndicatorContainer = styled(Animated.View)`
+  ${position.size('100%')};
+  position: absolute;
+  align-items: flex-end;
+  justify-content: flex-end;
+  ${padding(10)};
+`;
+
+export default function ModelViewer({
+  loading,
+  setLoading,
+  style,
+  uri,
+  alt,
+  fallbackUri,
+}: ModelViewerProps): JSX.Element {
+  const opacity = useMemo(() => new Animated.Value(1), []);
+  const [visibility, setVisibility] = useState<boolean>(false);
+  const [progress, setProgress] = useState<number>(0);
+  const originWhiteList = useMemo(() => ['*'], []);
+  const onMessage = useCallback(
+    ({ nativeEvent: { data } }) => {
+      const { type, payload } = JSON.parse(data);
+      if (type === 'progress') {
+        setProgress(payload as number);
+      } else if (type === 'model-visibility') {
+        setVisibility(payload as boolean);
+      }
+    },
+    [setProgress, setVisibility]
+  );
+  const source = useMemo(
+    () => ({
+      html: `
```

Comment:
please move this template somewhere

Reply:
@osdnk Why? This template only makes sense to this component and is tightly coupled to the logic in the `onMessage` callback.

Reply:
Because the formatting looks very weird.